copter/source/docs/systemid-mode-ctrl-optimization.rst

```diff
@@ -0,0 +1,257 @@
+
++------------------------------------+-----------------+---------------+----------------------+
+| Requirement Goal                   | Pitch & Roll    | Yaw           | Optimization Target  |
++------------------------------------+-----------------+---------------+----------------------+
+| Phase Margin                       | 50°             | 50°           | Larger is better     |
++------------------------------------+-----------------+---------------+----------------------+
+| Gain Margin                        | 7 dB            | 7 dB          | Larger is better     |
++------------------------------------+-----------------+---------------+----------------------+
+| Step Response: Response Time       | 0.02 s          | 0.2s          | Smaller is better    |
++------------------------------------+-----------------+---------------+----------------------+
+| Step Response: Overshoot           | 5 %             | 5 %           | Smaller is better    |
++------------------------------------+-----------------+---------------+----------------------+
+| Step Response: Stationary Error    | 5 %             | 0.1 %         | Smaller is better    |
++------------------------------------+-----------------+---------------+----------------------+
+| Control Variable Max. Overshoot    | 80 %            | 10 %          | Smaller is better    |
++------------------------------------+-----------------+---------------+----------------------+
+| Disturbance Rejection              | see graph below |not applied    | Smaller is better    |
++------------------------------------+-----------------+---------------+----------------------+
+
+The frequency-based disturbance rejection curve in decibel is shown below.
+As a reference, the blue line represents the rejection gain that is achieved with the default parameters of ArduCopter.
+The optimized system is required to stay bellow the yellow area to have a better disturbance rejection.
+
+.. image:: ../images/disturbanceRejectionRequirement.png
+:name: fig-dist-reject-req
+
+Optimization Results in the Simulation
+--------------------------------------
+
+The next four plots show the bode plot of the closed-loop system the simulation results for the tracking behavior of a 10 degree step, the normalized controller output corresponding to the angle step as well as the disturbance behavior for the roll axis.
+The normalized controller output evaluates whether the optimized behavior can actually be realized by the real system.
+It is assumed that the disturbances act as external torques on the airframe.
+Therefore, the disturbance behavior is simulated as a step added to the controller output signal.
+
+Bode plot of the closed-loop system:
+
+.. image:: ../images/rollAxisClosedLoopBode.png
+:name: fig-bode-cl-system
+
+Tracking Behaviour:
+
+.. image:: ../images/rollAxisTrackingSim.png
```

Comment:
I find it surprising that the autotune produced a response that was so oscillatory.  I don't think that would be acceptable.  @lthall any thoughts why the autotune would have resulted with a tune that was so oscillatory?  I would think they could do better with the autotune feature.

Reply:
In my opinion this is the main problem with this page. The response of autotune is obviously not based on a good autotune OR the system identification is incorrect.
The results of a successful system identification and optimisation should result in tuning values similar to autotune based on your requirements.
A good autotune should have responses similar to your optimised response. (autotune generates step responses and adjusts gains based on overshoot and the first minimum so it is obvious that the responses shown here are inconsistent with autotune)

Reply:
The simulation results are based on an attitude controller that was simplified quite a bit since I didn't have time left at the end of my master thesis to realize a proper reconstruction of the controller structure. As an example, the acceleration limits were not implemented in order to linearize the controller for the model-based design. The improvement of the controller used for the simulation will be done in the upcoming work, so we can achieve more reliable simulation results. Additionally, we can provide plots that compare the real tracking behaviour of both controllers.

@lthall I don't fully agree that the requirements for the model-based tune should necessarily result in a similar tune. To my understanding, autotune is evaluating the resulting tune based on the tracking behaviour alone. Contrary to this, the model-based approach considers stability and disturbance behaviour as well. Since these are conflicting design targets, the resulting parameters are likely to differ from the autotune approach. Also, we allow different values for the P and I part of the controller, so the optimization has more freedom in finding good parameter values based on the given requirements.

Reply:
I said similar, not identical. The response should be very close too. You are using the same overshoot requirements and a fast step response time requirement. These two goals will pull the result towards the autotuned result. Phase and gain margin may be a little different to the autotune result but I am guessing that it will be similar (this is one of the things I am interested in seeing how they vary). Again, disturbance rejection isn't going to be pushing anything to far away from the autotuned result.

Step response stationary error should be zero unless I don't understand what that is. I am not confident I do understand what this is. (all errors should decay to zero)
I don't know what "Control Variable Max. Overshoot" is.

The only significant difference I can see between the two will be the different requirement for the I term. Again, this is one of the things I am interested to see how they differ.